Sources/BraintreePaymentFlow/Public/BraintreePaymentFlow/BTPaymentFlowClient.h

```diff
@@ -136,7 +136,7 @@ typedef NS_ENUM(NSInteger, BTPaymentFlowErrorType) {
  @param request A BTPaymentFlowRequest request.
  @param completionBlock This completion will be invoked exactly once when the payment flow is complete or an error occurs.
  */
-- (void)startPaymentFlow:(BTPaymentFlowRequest<BTPaymentFlowRequestDelegate> *)request completion:(void (^)( BTPaymentFlowResult * _Nullable result,  NSError * _Nullable error))completionBlock;
+- (void)startPaymentFlow:(NSObject<BTPaymentFlowRequestDelegate> *)request completion:(void (^)( NSObject * _Nullable result,  NSError * _Nullable error))completionBlock;
```

Comment:
I don't think using `NSObject` in the merchant-facing API is expressive enough.

Reply:
For example:

The Swift version of this func would be:  
```
func startPaymentFlow(_ request: NSObjectProtocol & BTPaymentFlowRequestDelegate, completion completionBlock: @escaping (NSObject?, Error?) -> Void)
```
---
An example of the Xcode auto-fill when in Swift:
![Screen Shot 2023-02-03 at 10 09 29 AM](https://user-images.githubusercontent.com/35243507/216651309-2da3e0ce-a641-4a9e-ae13-8268bd37bfc0.png)

Reply:
That's a good point - the other option would be to override this method (or call the super method with the more generic interface) in the individual client implementations. The main advantage being that it would be even more strongly typed than it is now - so for 3DS the merchant facing API interface could _only_ use and return BTThreeDSecureRequest/Result vs even what we have today using BTPaymentFlowRequest/Result. 

It will ultimately require moving some things around throughout the conversion because we'd likely want a BTLocalPaymentMethodClient and to create a BTThreeDSecureClient (vs a category that we have now). That seems to be our preference though in other module to keep the merchant facing API more strongly typed than it is today, if possible. This is a similar approach to what we took in the PayPal/PayPalNative modules with the tokenize methods. Thoughts?

Reply:
I started spiking what I was thinking above out and it's gonna be a nightmare and a half. Technically possible but will require more changes than I think it's worth. Especially since it'd involve breaking changes (while small but still breaking) to 3DS, LPMs, and PaymentFlow. My thoughts are we should scrap converting PaymentFlow for now and make it dependent on converting 3DS first as that will minimizing the breaking changes in all payment methods that depend on PaymentFlow. Then just move forward with:
1. Convert another small module that has no dependancies to get us to the "halfway point"
2. Still switch payment flow to use ASWeb as this will get us in a good place down the line for the future conversion of modules in v7

Curious to hear your thoughts @scannillo!

Reply:
> That seems to be our preference though in other module to keep the merchant facing API more strongly typed than it is today, if possible.

I agree. Currently, merchants need to cast the result type, which is a little odd.

```
 self.paymentFlowDriver.startPaymentFlow(request) { (result, error) in
    guard let result = result as? BTThreeDSecureResult else {
        // Handle error
    }
}
```

Reply:
I think that sounds like an excellent plan forward! Thank you for the thorough investigation on this 👏